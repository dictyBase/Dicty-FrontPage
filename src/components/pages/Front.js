// @flow
import React from "react"
import { Flex, Box } from "rebass"
import Slideshow from "components/frontpage/Slideshow"
import News from "components/frontpage/News"
import Papers from "components/frontpage/Papers"
import Popular from "components/frontpage/Popular"
import Annotations from "components/frontpage/Annotations"
import StockCenter from "components/frontpage/StockCenter"
import papers from "data/papers"
import widgets from "data/widgets"
import news from "data/news"
import annotations from "data/annotations"
import stockcenter from "data/stockcenter"
import images from "data/carouselimages"

/** This is the frontpage component that appears when the user hits the "/" route. */

const Front = () => {
  return (
    <div>
      <Flex wrap mx={30}>
        <Box w="100%">
          <Flex wrap>
            <Box px={16} py={8} width={["100%", "100%", "100%", "50%"]}>
              <Slideshow images={images} />
            </Box>
            <Box px={16} py={8} width={["100%", "100%", "100%", "50%"]}>
              <News posts={news.slice(0, 5)} />
            </Box>
          </Flex>
          <Flex wrap>
            <Box px={2} py={1} width={["100%", "100%", "100%", "50%"]}>
              <Papers papers={papers.slice(0, 5)} />
            </Box>
            <Box px={2} py={1} width={["100%", "50%", "50%", "25%"]}>
              <Popular widgets={widgets} />
            </Box>
            <Box px={2} py={1} width={["100%", "50%", "50%", "25%"]}>
              <StockCenter stockcenter={stockcenter} />
              <Annotations annotations={annotations} />
            </Box>
          </Flex>
        </Box>
      </Flex>
    </div>
  )
}

export default Front
