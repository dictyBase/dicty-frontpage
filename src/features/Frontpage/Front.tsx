import React from "react"
import { Helmet } from "react-helmet"
import Grid from "@material-ui/core/Grid"
import { makeStyles, Theme } from "@material-ui/core/styles"
import Slideshow from "./Slideshow"
import Tweets from "./Tweets"
import Papers from "./Papers"
import Popular from "./Popular"
import Annotations from "./Annotations"
import StockCenter from "./StockCenter"
import papers from "common/data/papers"
import annotations from "common/data/annotations"
import stockcenter from "common/data/stockcenter"

const useStyles = makeStyles((theme: Theme) => ({
  container: {
    paddingLeft: "30px",
    paddingRight: "30px",
    [theme.breakpoints.up("xl")]: {
      paddingLeft: 0,
      paddingRight: 0,
    },
  },
  topItem: {
    padding: "8px 16px 8px 16px",
  },
  bottomItem: {
    padding: "1px 2px 1px 2px",
  },
  tweetBox: {
    padding: "8px 16px 8px 16px",
    width: "100%",
  },
}))

/** This is the frontpage component that appears when the user hits the "/" route. */

const Front = () => {
  const classes = useStyles()

  return (
    <div>
      <Helmet>
        <title>
          dictyBase - your central resource for Dictyostelid genomics
        </title>
        <meta
          name="description"
          content="dictyBase is a central resource for Dictyostelid genomics"
        />
      </Helmet>
      <Grid container justify="center" className={classes.container}>
        <Grid item className={classes.topItem} sm={12} md={6} xl={5}>
          <Slideshow />
        </Grid>
        <Grid item className={classes.tweetBox} sm={12} md={6} xl={5}>
          <Tweets />
        </Grid>
        <Grid item className={classes.topItem} xs={12} lg={6} xl={4}>
          <Papers papers={papers.slice(0, 4)} />
        </Grid>
        <Grid item className={classes.topItem} xs={12} sm={6} lg={3}>
          <Popular />
        </Grid>
        <Grid item className={classes.topItem} xs={12} sm={6} lg={3}>
          <StockCenter stockcenter={stockcenter} />
          <Annotations annotations={annotations} />
        </Grid>
      </Grid>
    </div>
  )
}

export default Front
