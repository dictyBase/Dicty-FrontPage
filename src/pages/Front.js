// @flow
import React, { Component } from "react"
import { Flex, Box } from "grid-styled"
import styled from "styled-components"
import Slideshow from "../Components/Slideshow"
import News from "../Components/News"
import Papers from "../Components/Papers"
import Popular from "../Components/Popular"
import Annotations from "../Components/Annotations"
import StockCenter from "../Components/StockCenter"
import news from "../data/news"
import papers from "../data/papers"
import widgets from "../data/widgets"
import annotations from "../data/annotations"
import stockcenter from "../data/stockcenter"


const images = [
  "https://i.ytimg.com/vi/uXdzuz5Q-hs/maxresdefault.jpg",
  "https://i.ytimg.com/vi/uXdzuz5Q-hs/maxresdefault.jpg",
  "https://i.ytimg.com/vi/uXdzuz5Q-hs/maxresdefault.jpg",
  "https://i.ytimg.com/vi/uXdzuz5Q-hs/maxresdefault.jpg",
]

const Container = styled.div`
  display: flex;
  flex-direction: row;
  height: 50%;
  width: 90%;
  margin: auto;

  @media (max-width: 767px){
    flex-wrap: wrap;
  }
`
const Item = styled.div`
  height: 400px;
  padding: 8px 16px 8px 16px;  
`

// width: 50%;
// @media (max-width: 767px){
//   width: 100%;
// }

class Front extends Component {
  render() {
    return (
      <div>
        <Flex>
          <Container>
              <Box px={16} py={8} width={[1/2]}>
                <Slideshow images={images} />
              </Box>
              <Box px={16} py={8} width={[1/2]}>
                <News posts={news.slice(0, 3)} />
              </Box>
          </Container>
        </Flex>
        <Flex>
          <Container>
            <Box px={2} py={1} width={[1/2]}>
                <Papers papers={papers.slice(0, 5)} />
            </Box> 
            <Box px={2} py={1} width={[1/4]}>
                  <Popular widgets={widgets}/>
            </Box> 
            <Box px={2} py={1} width={[1/4]}>
                  <StockCenter stockcenter={stockcenter}/> 
                  <Annotations annotations={annotations}/>  
            </Box> 
          </Container>            
        </Flex>
      </div>
    )
  }
}

export default Front

