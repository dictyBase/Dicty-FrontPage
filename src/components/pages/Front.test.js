import React from "react"
import toJson from "enzyme-to-json"
import { shallow } from "enzyme"
import "jest-styled-components"
import Front from "./Front"

test("matching a snapshot of Front", () => {
  const subject = shallow(<Front />)
  expect(toJson(subject)).toMatchSnapshot()
})
