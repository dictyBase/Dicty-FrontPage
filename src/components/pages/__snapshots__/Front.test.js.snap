// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matching a snapshot of Front 1`] = `
<div>
  <Flex
    mx={30}
    wrap={true}
  >
    <Box
      px={16}
      py={8}
      width={
        Array [
          "100%",
          "100%",
          "50%",
          "50%",
        ]
      }
    >
      <Slideshow />
    </Box>
    <Box
      px={16}
      py={8}
      width={
        Array [
          "100%",
          "100%",
          "50%",
          "50%",
        ]
      }
    >
      <News
        posts={
          Array [
            Object {
              "content": "Dicty in the News: Bacteria carried as food within Dictyostelium discoideum are protected by a lectin coat. Reported in ScienceNews, and recently published by Dinh and colleagues in Science with comments by co-author Adam Kuspa.",
              "date": "2018.07.21",
              "source": "ScienceNews",
            },
            Object {
              "content": "Dicty2018: A list of currently registered people is available here. Please check if all your information has been correctly processed and email the organizers if corrections are needed. ",
              "date": "2018.07.19",
              "source": "dictyBase",
            },
            Object {
              "content": "Thierry Soldati is amongst 62 life scientists elected to EMBO (European Molecular Biology Organisation) this year. New EMBO members. Congratulations Thierry!",
              "date": "2018.05.14",
              "source": "Dicty Stock Center",
            },
            Object {
              "content": "Thomas Williams, who graduated and now is a post-doc in Rob Kay's lab, recently first authored a paper in JCS. In addition, JCS chose him to be First Person, a series of interviews with selected first authors, helping early-career researchers promote themselves alongside their papers.",
              "date": "2018.04.24",
              "source": "Dicty Stock Center",
            },
            Object {
              "content": "After Northwestern moved our servers from one campus to the other recently our Blast from the Front - and all Gene Pages, as well as the Genome Browser (including snapshot on gene page) are down. We and Northwestern IT staff are working on the problem. Meanwhile for Blast use the Genome Server, the Blast server which is available from the other genome sites we house. Thanks for your patience! ",
              "date": "2018.02.13",
              "source": "dictyBase",
            },
          ]
        }
      />
    </Box>
    <Box
      px={2}
      py={1}
      width={
        Array [
          "100%",
          "100%",
          "100%",
          "50%",
        ]
      }
    >
      <Papers
        papers={
          Array [
            Object {
              "author": "Usaj, Zattelman, Regev, Shneyer, Wiesel-Motiuk & Henn",
              "journal": "Anal Biochem. 2018 Jul 31. pii: S0003-2697(18)30419-6. doi: 10.1016/j.ab.2018.07.026. [Epub ahead of print]",
              "link": "https://www.ncbi.nlm.nih.gov/pubmed/30075102",
              "title": "Overexpression and purification of human myosins from transiently and stably transfected suspension adapted HEK293SF-3F6 cells.",
            },
            Object {
              "author": "Marinović M, Xiong H, Rivero F, Weber I.",
              "journal": "Methods Mol Biol. 2018;1821:371-392. doi: 10.1007/978-1-4939-8612-5_25.",
              "link": "https://www.ncbi.nlm.nih.gov/pubmed/30062425",
              "title": "Assaying Rho GTPase-Dependent Processes in Dictyostelium discoideum.",
            },
            Object {
              "author": "Tunnacliffe E, Corrigan AM, Chubb JR.",
              "journal": "Proc Natl Acad Sci U S A. 2018 Jul 30. pii: 201800943. doi: 10.1073/pnas.1800943115. [Epub ahead of print]",
              "link": "https://www.ncbi.nlm.nih.gov/pubmed/30061408",
              "title": "Promoter-mediated diversification of transcriptional bursting dynamics following gene duplication.",
            },
            Object {
              "author": "Song L, Rijal R, Karow M, Stumpf M, Hahn O, Park L, Insall R, Schröder R, Hofmann A, Clemen CS, Eichinger L.",
              "journal": "Dis Model Mech. 2018 Jul 30. pii: dmm.033449. doi: 10.1242/dmm.033449. [Epub ahead of print]",
              "link": "https://www.ncbi.nlm.nih.gov/pubmed/30061306",
              "title": "Expression of N471D strumpellin leads to defects in the endolysosomal system.",
            },
          ]
        }
      />
    </Box>
    <Box
      px={2}
      py={1}
      width={
        Array [
          "100%",
          "50%",
          "50%",
          "25%",
        ]
      }
    >
      <Popular
        widgets={
          Array [
            Object {
              "alt": "techniques",
              "image": "widgetTECHNIQUES.png",
              "isRouter": true,
              "link": "/research/techniques",
            },
            Object {
              "alt": "genome browser",
              "image": "widgetGBROWSER.png",
              "isRouter": false,
              "link": "http://dictybase.org/tools/jbrowse/?data=data/jbrowse/discoideum&loc=6:1..50011&tracks=reference,gene,transcript",
            },
            Object {
              "alt": "learn dicty",
              "image": "widgetLEARNDICTY.png",
              "isRouter": true,
              "link": "/explore/learn",
            },
            Object {
              "alt": "dicty STATS (dictyAccess)",
              "image": "widgetSTATS.png",
              "isRouter": true,
              "link": "/dictyaccess",
            },
            Object {
              "alt": "jobs @ dicty",
              "image": "widgetJOBS.png",
              "isRouter": true,
              "link": "/community/jobs",
            },
            Object {
              "alt": "dicty express",
              "image": "widgetDICTYEXPRESS.png",
              "isRouter": false,
              "link": "https://dictyexpress.research.bcm.edu/landing/",
            },
          ]
        }
      />
    </Box>
    <Box
      px={2}
      py={1}
      width={
        Array [
          "100%",
          "50%",
          "50%",
          "25%",
        ]
      }
    >
      <StockCenter
        stockcenter={
          Object {
            "plasmids": Array [
              "DBS0304455",
              "DSB0350408",
              "DBS0350326",
            ],
            "strains": Array [
              "DBS0304455",
              "DSB0350408",
              "DBS0350326",
              "DBS0350435",
            ],
          }
        }
      />
      <Annotations
        annotations={
          Object {
            "genes": Array [
              "abprtA",
              "trgrB1",
              "tgrC1",
              "tacA",
            ],
            "papers": Array [
              "25609090",
              "25546705",
              "25540127",
              "25596489",
            ],
          }
        }
      />
    </Box>
  </Flex>
</div>
`;
